src/BDCS/API/Server.hs

```diff
@@ -174,11 +187,32 @@ mkApp bdcsPath gitRepoPath sqliteDbPath = do
     -- client uses a different route to check and fetch the results.
     void $ async $ composeServer cfg
 
-    return $ app cfg
+    return $ application cfg
 
 -- | Run the API server
-runServer :: Int -> FilePath -> FilePath -> FilePath -> IO ()
-runServer port bdcsPath gitRepoPath sqliteDbPath = run port =<< mkApp bdcsPath gitRepoPath sqliteDbPath
+runServer :: FilePath -> FilePath -> FilePath -> FilePath -> IO ()
+runServer socketPath bdcsPath gitRepoPath sqliteDbPath = void $ withSocketsDo $ do
+    sock <- getSocket socketPath
+    app  <- mkApp bdcsPath gitRepoPath sqliteDbPath
+    runSettingsSocket defaultSettings sock app
+ where
+    getSocket :: FilePath -> IO Socket
+    getSocket fp = lookupEnv "LISTEN_FDS" >>= \case
+        Nothing -> if fp == "" then CES.throw NoSocketError else newSocket fp
+        Just s  -> case readMaybe s of
+            Nothing -> CES.throw BadFileDescriptor
+            Just fd -> mkSocket fd AF_UNIX Stream defaultProtocol Bound
+
+    newSocket :: FilePath -> IO Socket
+    newSocket path = do
+        whenM (doesPathExist path) $
+            removePathForcibly path
+
+        s <- socket AF_UNIX Stream defaultProtocol
+        bind s (SockAddrUnix path)
+        listen s 1
+        setFileMode path 0o660
+        return s
```

Comment:
We also need to set ownership to root:weldr (or whatever group is passed in by --group). This is how we control access to it, the user who logs in to cockpit has to either be root or a user who is a member of the correct group.